### DISCUSSION TITLE
Web UI

### DISCUSSION THREAD

**Discussion by weiwangorg** [smly/mjai.app#155] [Ideas]
I used the game replay provided by the Web UI and found it very useful.

Would you open source the web UI? I feel that using the web UI to replay game logs would be more intuitive and easier to understand, and it would also conveniently allow human-model battles to better perceive the differences between bots and humans.

**Comment by smly** [comment]
Thanks for the feedback!

I'm definitely motivated to open-source the Web UI. However, I'm not entirely satisfied with the current state of the Web UI and would like to improve it before release. Specifically, I'm aiming to add features like displaying yaku and han information for winning hands, as well as supporting drag-and-drop of mjai format JSON files to initiate replays, inspired by the Web UI in [https://github.com/Lux-AI-Challenge/Lux-Design-S3](https://www.google.com/url?sa=E&q=https%3A%2F%2Fgithub.com%2FLux-AI-Challenge%2FLux-Design-S3). This is likely a longer-term project, but it's something I'm working towards.

Of course, any help would be greatly appreciated! :-)
